content/terminal/custom-backend/Widgets/metric.md

```diff
@@ -0,0 +1,89 @@
+---
+title: Metric Example
+sidebar_position: 3
+description: Learn how to integrate your own backend with OpenBB Terminal Pro using the cookie-cutter or language-agnostic API approaches, with illustrative guides and principles for handling widget.json files, APIs, interfaces, Python, FastAPI, and more.
+keywords:
+- widgets.json
+- OpenBB API
+- Endpoint integration
+- widget configuration
+- Language-Agnostic API
+- API implementation
+- Python
+- FastAPI
+- Terminal Pro widgets
+- Widget definitions
+---
+
+# Introduction
+
+This guide will walk you through the process of creating a metric widget for OpenBB Terminal Pro. By the end of this guide, you will have a working metric widget that you can add to OpenBB Terminal Pro.
+
+> **Prerequisite:** Ensure you have Python > 3.11 installed on your system before proceeding.
+
+## Step 1: Set Up Your Project
+
+To get started, create the main application file and the widget configuration file. You will only need these two files:
+
+- `main.py`: This file will contain your FastAPI application code.
+- `widgets.json`: This file will define the configuration for your widget.
+
+## Step 2: Create the Backend
+
+### Edit the Main Application File
+
+Edit the `main.py` file and add the following code. This code sets up a FastAPI application with an endpoint to serve the widget configuration and data for a metric:
+
+```python
+...
+# Metric endpoint
+@app.get("/test_metric")
+def test_metric():
+    """Example endpoint to provide metric data."""
+    # Example data structure
```

Comment:
`# Example data structure` -> `# Example data`

The data structure is the same all the time for this widget. It's the data itself that changes